sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py

```diff
@@ -63,17 +66,18 @@ def _any_conditions(modified_access_conditions=None, **kwargs):  # pylint: disab
 
 
 def upload_block_blob(  # pylint: disable=too-many-locals, too-many-statements
-        client=None,
-        data=None,
-        stream=None,
-        length=None,
-        overwrite=None,
-        headers=None,
-        validate_content=None,
-        max_concurrency=None,
-        blob_settings=None,
-        encryption_options=None,
-        **kwargs):
+    client: "BlockBlobOperations" = None,  # type: ignore [assignment]
+    data: Union[bytes, Iterable[AnyStr], IO[AnyStr]] = None,  # type: ignore [assignment]
+    overwrite: bool = None,  # type: ignore [assignment]
+    encryption_options: Dict[str, Any] = None,  # type: ignore [assignment]
+    blob_settings: "StorageConfiguration" = None,  # type: ignore [assignment]
+    headers: Dict[str, Any] = None,  # type: ignore [assignment]
+    stream: IO = None,  # type: ignore [assignment]
+    length: Optional[int] = None,
+    validate_content: bool = None,  # type: ignore [assignment]
+    max_concurrency: Optional[int] = None,  # type: ignore [assignment]
+    **kwargs: Any
+) -> Dict[str, Any]:
```

Comment:
Since this is a private method and there is only one caller, go ahead and change this method signature to be non-optional for things that we always pass and remove all ignores. (Like you did for `upload_*_blob` below)